### PR TITLE
Added user restriction

### PR DESCRIPTION
**Pull request by satyamamit**
This PR fixes the issues which is for user restrictions not visible on link analysis page.

image

### CODE REVIEW

File: backend/queues/src/main/java/com/griddynamics/msd365fp/manualreview/queues/service/PublicLinkAnalysisService.java [hunk 1]
```diff
@@ -265,15 +265,12 @@ public LinkAnalysisDTO createLinkAnalysisEntry(final LinkAnalysisCreationDTO req
             }
         }
 
-        if (checkUserRestriction) {
             result.forEach(details -> {
                 if (details.getUser() != null &&
                         details.getUser().getEmail() != null) {
-                    UserEmailListEntity userEmailLists = dfpExplorerService.exploreUserEmailList(details.getUser().getEmail());
-                    details.setUserRestricted(userEmailLists.getCommonRestriction());
+                    details.setUserRestricted(this.isUserFraud(details.getUser().getEmail()));
```
Comment (irinatarn) [inline backend/queues/src/main/java/com/griddynamics/msd365fp/manualreview/queues/service/PublicLinkAnalysisService.java:271]: I don't think this is correct.
Previous code checked if the user is on safe or block list so I think we need to preserve that.
You can do this.isUserFraud(details.getUser().getEmail()) || userEmailLists.getCommonRestriction()) here.
Or we can modify the UX to have 2 columns - restricted from label and restricted from Block/Safe list.
I think the first option is faster, but the second is more explicit - your choice.

Comment (satyamamit) [reply backend/queues/src/main/java/com/griddynamics/msd365fp/manualreview/queues/service/PublicLinkAnalysisService.java:271]: how are you doing it at other places in console, if you are doing only with Label. we should only do it with labels only.

Comment (irinatarn) [reply backend/queues/src/main/java/com/griddynamics/msd365fp/manualreview/queues/service/PublicLinkAnalysisService.java:271]: Nope - we have different fields in the transactions for that. Let me call you and show.

**Review by irinatarn — APPROVED** [review]

File: backend/queues/src/main/java/com/griddynamics/msd365fp/manualreview/queues/service/PublicLinkAnalysisService.java (hunk 1 above)
Comment (satyamamit) [reply backend/queues/src/main/java/com/griddynamics/msd365fp/manualreview/queues/service/PublicLinkAnalysisService.java:271]: as discussed, closing this comment.

**Pull request merged**